### PR TITLE
Examples custom nav guide

### PR DESCRIPTION
**Pull request by gwyneplaine**
Guide and example for #6111 
Split out for expedited functional release.

### CODE REVIEW

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  | { state: 'error'; error: Error | readonly [GraphQLError, ...GraphQLError[]] };
+```
+
+This should be handled for you automatically by passing the prop into the `NavigationContainer`,
+
+!> `NavigationContainer` is one of many assistive components to help you more easily compose your custom Navigation component. You can read more about it down [below](#components).
+
+?> If you wish to render out the `authenticatedItem` in a custom manner, please be aware that you will need to handle all of the possible states of the `authenticatedItem`. Explaining how to do this is out of the scope of this particular guide.
+
+### Components
```
Comment (gwyneplaine) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:51]: Honestly this is definitely not the right place for *this* kind of documentation. 

Very much taking a 'let's get this content in first and figure out where it goes later' approach here.

File: docs/components/docs/ExamplesList.tsx [hunk 2]
```diff
@@ -140,6 +140,14 @@ export function Examples() {
       >
         Adds a custom logo component in the Admin UI. Builds on the Task Manager starter project.
       </Well>
+      <Well
+        grad="grad2"
+        heading="Custom Admin-UI Navigation"
```
Comment (JedWatson) [inline docs/components/docs/ExamplesList.tsx:145]:
```suggestion
        heading="Custom Admin UI Navigation"
```

Comment (JedWatson) [reply docs/components/docs/ExamplesList.tsx:145]: For consistency with other navigation items:

image

File: docs/components/docs/Navigation.tsx [hunk 3]
```diff
@@ -155,6 +155,9 @@ export function DocsNavigation() {
         <NavItem href="/docs/guides/custom-admin-ui-pages">
           Custom Admin UI Pages <Badge look="success">New</Badge>
         </NavItem>
+        <NavItem href="/docs/guides/custom-admin-ui-navigation">
+          Custom Admin-UI Navigation <Badge look="success">New</Badge>
```
Comment (JedWatson) [inline docs/components/docs/Navigation.tsx]:
```suggestion
          Custom Admin UI Navigation <Badge look="success">New</Badge>
```

For consistency with other nav items

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 4]
```diff
@@ -0,0 +1,237 @@
+import { Markdown } from '../../../components/Markdown';
+
+# Custom Admin UI Navigation
+
+In this guide, we'll illustrate how to create a custom Navigation component with custom routes to be rendered in the Admin-UI.
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:5]:
```suggestion
In this guide, we'll show you how to create a custom Navigation component with custom routes to be rendered in the Admin-UI.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+type AuthenticatedItem =
+  | { state: 'unauthenticated' }
+  | { state: 'authenticated'; label: string; id: string; listKey: string }
+  | { state: 'loading' }
+  | { state: 'error'; error: Error | readonly [GraphQLError, ...GraphQLError[]] };
+```
+
+This should be handled for you automatically by passing the prop into the `NavigationContainer`,
+
+!> `NavigationContainer` is one of many assistive components to help you more easily compose your custom Navigation component. You can read more about it down [below](#components).
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:47]:
```suggestion
!> `NavigationContainer` is one of several higher-level components to help you more easily compose your custom Navigation component. You can read more about it down [below](#components).
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+
+```ts
+type AuthenticatedItem =
+  | { state: 'unauthenticated' }
+  | { state: 'authenticated'; label: string; id: string; listKey: string }
+  | { state: 'loading' }
+  | { state: 'error'; error: Error | readonly [GraphQLError, ...GraphQLError[]] };
+```
+
+This should be handled for you automatically by passing the prop into the `NavigationContainer`,
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:45]:
```suggestion
The `authenticatedItem` props is rendered automatically when you pass it into the `NavigationContainer` component.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+
+!> `NavigationContainer` is one of many assistive components to help you more easily compose your custom Navigation component. You can read more about it down [below](#components).
+
+?> If you wish to render out the `authenticatedItem` in a custom manner, please be aware that you will need to handle all of the possible states of the `authenticatedItem`. Explaining how to do this is out of the scope of this particular guide.
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:49]:
```suggestion
?> If you render the `authenticatedItem` yourself, make sure you handle all of the possible states.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+
+### Components
+
+Keystone exposes a variety of helper components to make building out your custom Admin-UI Navigation component easier. These are:
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:53]:
```suggestion
Keystone exposes a variety of helper components to make building out your custom Admin UI Navigation component easier. These are:
```

"Admin UI" is never "Admin-UI"

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+
+- [NavigationContainer](#navigation-container)
+- [ListNavItems](#list-nav-items)
+- [ListNavItem](#list-nav-item)
+- [NavItem](#nav-item)
+
+#### NavigationContainer
+
+This is component renders containing markup around your routes, as well as as the different stats of the authenticatedItem prop.
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:62]:
```suggestion
This component renders containing markup around your navigation links, as well as as the different states of the `authenticatedItem` prop.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+
+```tsx
+import { NavigationContainer} from '@keystone-next/keystone/admin-ui/components'
+```
+
+#### ListNavItems
+
+This component takes the provided Array of `lists` and renders out a list of NavItems.
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:70]:
```suggestion
This component takes the provided Array of `lists` and renders a list of NavItems.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 11]
```diff
@@ -0,0 +1,237 @@
...
+```tsx
+import { ListNavItems } from '@keystone-next/keystone/admin-ui/components'
+```
+
+As mentioned above the routes object has the following type:
+
+```typescript
+type lists = ListMeta[]
+type include = string[]
+```
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx]: This seems incorrect, maybe from an older version of the API?

The "routes object" isn't actually mentioned above (the concept of routes is) and these are just props, per the example below.

Better to express this (and the two types below) as describing props:

```suggestion
The `ListNavItems` component takes the following props:

```typescript
type ListNavProps = {
  lists: ListMeta[];
  include: string[];
}
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 12]
```diff
@@ -0,0 +1,237 @@
...
+
+It expects the following props.
+
+```typescript
+type ListNavItemProps = {
+  list: ListMeta;
+};
+```
+
+The impetus behind `ListNavItem` is to have a higher-level abstraction that can allow users to not have to care about the contents and structure of our lists.
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:124]:
```suggestion
The idea behind `ListNavItem` is to have a higher-level abstraction that makes it easier to render navigation links for lists consistently.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 13]
```diff
@@ -0,0 +1,237 @@
...
+This is especially relevant for future feature additions to the nav. (i.e. if we add icon support, that would work out of the box from a dependency bump for users of ListNavItem, whereas you'd have to add that icon prop manually leveraging a lower level component.)
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:125]:
```suggestion
This is especially relevant for future feature additions to the nav. (i.e. if we add icon support, that would work out of the box from a dependency bump for users of ListNavItem, whereas you'd have to add that icon prop manually with a lower level component.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 14]
```diff
@@ -0,0 +1,237 @@
...
+}
+```
+
+By default the `isSelected` value if left undefined, will be evaluated by the condition `router.pathname === href`, pass in `isSelected` if you have a custom condition or would like more granular control over the selected state of Navigation items.
+
+### Piecing it all together
+
+The code snippet below shows how to use these three components in conjunction to create a custom Navigation component that:
+
+a) Has the styling constraints of the default Admin-UI navigation
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:157]:
```suggestion
a) Has the styling constraints of the default Admin UI navigation
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 15]
```diff
@@ -0,0 +1,237 @@
...
+type ListNavItemProps = {
+  list: ListMeta;
+};
+```
+
+The impetus behind `ListNavItem` is to have a higher-level abstraction that can allow users to not have to care about the contents and structure of our lists.
+This is especially relevant for future feature additions to the nav. (i.e. if we add icon support, that would work out of the box from a dependency bump for users of ListNavItem, whereas you'd have to add that icon prop manually leveraging a lower level component.)
+
+Use this component when you want to have more granular control over rendering nav items from lists prop provided to the `Navigation` component by keystone,
+but would still like to not have to care about which list properties need to be associated with each NavItem prop.
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:128]:
```suggestion
but would still like to not have to worry about which list properties need to be associated with each NavItem prop.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 16]
```diff
@@ -0,0 +1,237 @@
...
+
+The code snippet below shows how to use these three components in conjunction to create a custom Navigation component that:
+
+a) Has the styling constraints of the default Admin-UI navigation
+
+b) Renders out all the necessary routes generated from our Keystone `lists`
+
+c) Adds a new custom route to take us to the `keystonejs docs`.
+
+d) Renders out additional UI based on whether or not the user has signed in provided by the `authenticatedItem` prop.
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:163]:
```suggestion
d) Renders additional UI based on whether or not the user has signed in provided by the `authenticatedItem` prop.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 17]
```diff
@@ -0,0 +1,237 @@
...
+      <NavItem href="https://keystonejs.com/guides/custom-navigation-guide">
+        Navigation Guide
+      </NavItem>
+    </NavigationContainer>
+  )
+}
+
+```
+
+## Adding the custom navigation component to the Admin-UI
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:211]:
```suggestion
## Adding the custom navigation component to the Admin UI
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 18]
```diff
@@ -0,0 +1,237 @@
...
+
+Keystone is made aware of custom components it needs to render in the Admin-UI via the `/admin/config.ts` file in your project root.
```
Comment (JedWatson) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:213]:
```suggestion
Keystone is made aware of custom components it needs to render in the Admin UI via the `/admin/config.ts` file in your project root.
```

File: examples/custom-admin-ui-navigation/README.md [hunk 19]
```diff
@@ -0,0 +1,161 @@
+## Feature Example - Custom Field View
+
+This project demonstrates how to create a custom Navigation component for use in the Admin-UI. It builds on the [Task Manager](../task-manager) starter project.
```
Comment (JedWatson) [inline examples/custom-admin-ui-navigation/README.md:3]:
```suggestion
This project demonstrates how to create a custom Navigation component for use in the Admin UI. It builds on the [Task Manager](../task-manager) starter project.
```

**Review by JedWatson — CHANGES_REQUESTED** [review]

File: examples/custom-admin-ui-navigation/README.md [hunk 20]
```diff
@@ -0,0 +1,162 @@
+## Feature Example - Custom Field View
```
Comment (JedWatson) [inline examples/custom-admin-ui-navigation/README.md:1]: This heading isn't right, this is the Admin UI Navigation example

**Review by JedWatson — DISMISSED** [review]

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 21]
```diff
@@ -0,0 +1,416 @@
...
+Keystone passes the following props to your custom Navigation component:
+
+```ts
+type NavigationProps {
+    lists: ListMeta[];
+    authenticatedItem: AuthenticatedItem;
+}
+```
+
+`lists` is an array of keystone list objects. Internally Keystone filters through your lists and ensures that only visible lists are passed through to you in the `lists` array prop.
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:189]:
```suggestion
`lists` is an array of Keystone list objects. Internally Keystone filters through your lists and ensures that only visible lists are passed through to you in the `lists` array prop.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 22]
```diff
@@ -0,0 +1,416 @@
...
+```typescript
+type ListNavItemProps = {
+  list: ListMeta;
+};
+```
+
+The idea behind `ListNavItem` is to have a higher-level abstraction that makes it easier to render navigation links for lists consistently.
+This is especially relevant for future feature additions to the nav. (i.e. if we add icon support, that would work out of the box from a dependency bump for users of ListNavItem, whereas you'd have to add that icon prop manually with a lower level component.)
+
+Use this component when you want to have more granular control over rendering nav items from lists prop provided to the `Navigation` component by keystone,
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:307]:
```suggestion
Use this component when you want to have more granular control over rendering nav items from lists prop provided to the `Navigation` component by Keystone,
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 23]
```diff
@@ -0,0 +1,416 @@
...
+  };
+};
+```
+
+Reference your custom Navigation component in the export as so, the exported component can either be declared inline or imported from elsewhere in your project.
+
+```ts
+import type { AdminConfig } from '@keystone-next/types';
+import { CustomNavigation } from './components/CustomNavigation';
+export  const component: AdminConfig['components'] = {
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:31]:
```suggestion
import { CustomNavigation } from './components/CustomNavigation';

export const component: AdminConfig['components'] = {
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 24]
```diff
@@ -0,0 +1,416 @@
...
+    Logo?: (props: {}) => ReactElement;
+    Navigation?: (props: NavigationProps) => ReactElement;
+  };
+};
+```
+
+Reference your custom Navigation component in the export as so, the exported component can either be declared inline or imported from elsewhere in your project.
+
+```ts
+import type { AdminConfig } from '@keystone-next/types';
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:29]:
```suggestion
```ts
// /admin/config.ts
import type { AdminConfig } from '@keystone-next/types';
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 25]
```diff
@@ -0,0 +1,416 @@
...
+Below is the expected type for the component export:
+
+```ts
+export type AdminConfig = {
+  components?: {
+    Logo?: (props: {}) => ReactElement;
+    Navigation?: (props: NavigationProps) => ReactElement;
+  };
+};
+```
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:24]:
```suggestion
```

This info isn't really helpful at this stage. We can just show them what to put in file, they can worry about typescript later.

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 26]
```diff
@@ -0,0 +1,416 @@
...
+# Custom Admin UI Navigation
+
+In this guide, we'll show you how to create a custom Navigation component to be rendered in the Admin UI.
+
+!> By the end of this guide you should have a custom Navigation component rendering in the Admin UI with a custom route pointing to the KeystoneJS docs.
+![finished custom navigation example](/assets/guides/custom-admin-ui-navigation/final-result.png)
+
+## Adding the custom navigation component to the Admin UI
+
+The first thing you'll need to do in creating your custom Navigation component is to specify an `/admin/config.ts` file in the root of your Keystone project.
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:12]:
```suggestion
The first thing you'll need to do is to specify an `/admin/config.ts` file in the root of your Keystone project.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 27]
```diff
@@ -0,0 +1,416 @@
...
+```ts
+export type AdminConfig = {
+  components?: {
+    Logo?: (props: {}) => ReactElement;
+    Navigation?: (props: NavigationProps) => ReactElement;
+  };
+};
+```
+
+Reference your custom Navigation component in the export as so, the exported component can either be declared inline or imported from elsewhere in your project.
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:26]:
```suggestion
Reference your custom Navigation component in the export as so.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 28]
```diff
@@ -0,0 +1,416 @@
...
+
+## Creating your custom Navigation component
+
+The next step is to create our `CustomNavigation` React component.
+In the root of your project create a new tsx file at `/admin/components/CustomNavigation.tsx`.
+
+Here we'll export our CustomNavigation component as a named export.
+
+```tsx
+export function CustomNavigation({ authenticatedItem, lists }: NavigationProps) {
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:44]: Let's add the appropriate `NavigationProps` import to this snippet, and also include a filename comment at the top for consistency.

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 29]
```diff
@@ -0,0 +1,416 @@
...
+  return (
+    {/* ... */}
+  )
+}
+```
+
+If it detects a Navigation property in the `config.tsx` file we created earlier, Keystone will pass the following props to this component.
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:51]:
```suggestion
Keystone will pass the following props to this component.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 30]
```diff
@@ -0,0 +1,416 @@
...
+
+### Setting up the layout
+
+Next we'll want to import some components that Keystone provides to help us build our custom Navigation.
+
+```tsx
+import { NavigationContainer, NavItem, ListNavItems } from '@keystone-next/keystone/admin-ui/components';
+```
+
+The `NavigationContainer` component renders containing markup around your navigation links, as well as as the different states of the `authenticatedItem` prop. We'll need this to:
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:70]:
```suggestion
The `NavigationContainer` component provides a container around your navigation links, as well as the different states of the `authenticatedItem` prop. We'll need this to:
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 31]
```diff
@@ -0,0 +1,416 @@
...
+    </NavigationContainer>
+  )
+}
+```
+
+!> For more information on the `ListNavItems` component, see the [ListNavItems](#list-nav-items) section of the [component docs](#components) below.
+
+### Adding additional routes
+
+The `NavItem` component is a thin styling and accessibility wrapper around the Link component from Next.js. We'll use this component to render out custom route as well as the `Dashboard` route that Keystone expects.
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:112]:
```suggestion
The `NavItem` component is a thin styling and accessibility wrapper around the Link component from Next.js. We'll use this component to render our custom route as well as the `Dashboard` route.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 32]
```diff
@@ -0,0 +1,416 @@
...
+        Navigation Guide
+      </NavItem>
+    </NavigationContainer>
+  )
+}
+```
+
+!> For more information on the `NavItem` component, see the [NavItem](#nav-item) section of the [component docs](#components) below.
+
+?> **Note** When opting into a custom Navigation component you will need to specify a NavItem for the `Dashboard` page (the index route). Keystone doesn't provide this to you out of the box, as it's not strictly a list concern, but doing so is straight-forward as seen in the snippet above.
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:133]:
```suggestion
?> **Note** When opting into a custom Navigation component you will need to specify a NavItem for the `Dashboard` page (the `/` route).
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 33]
```diff
@@ -0,0 +1,416 @@
...
+
+### Putting it all together
+
+With all that done, your Custom Navigation component should be good to go, and your admin folder should look like this.
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:137]:
```suggestion
With all that done, your Custom Navigation component should be good to go, and your `/admin` folder should look like this.
```

**Review by timleslie — CHANGES_REQUESTED** [review]

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 34]
```diff
@@ -0,0 +1,410 @@
...
+    Navigation: CustomNavigation
+}
+```
+
+## Creating your custom Navigation component
+
+The next step is to create our `CustomNavigation` React component.
+In the root of your project create a new tsx file at `/admin/components/CustomNavigation.tsx`.
+
+Here we'll export our CustomNavigation component as a named export.
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:32]:
```suggestion
Here we'll export our `CustomNavigation` component as a named export.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 35]
```diff
@@ -0,0 +1,410 @@
...
+The `NavItem` component is a thin styling and accessibility wrapper around the Link component from Next.js. We'll use this component to render our custom route as well as the `Dashboard` route.
+
+```tsx
+import { NavigationContainer, NavItem, ListNavItems } from '@keystone-next/keystone/admin-ui/components';
+import type { NavigationProps } from '@keystone-next/keystone/admin-ui/components';
+
+export function CustomNavigation({ authenticatedItem, lists }: NavigationProps) {
+  return (
+    <NavigationContainer authenticatedItem={authenticatedItem}>
+      <NavItem href="/">Dashboard<NavItem>
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:115]:
```suggestion
      <NavItem href="/">Dashboard</NavItem>
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 36]
```diff
@@ -0,0 +1,410 @@
...
+
+
+// admin/config.ts
+import { AdminConfig } from '@keystone-next/types';
+export type AdminConfig = {
+  components?: {
+    Logo?: (props: {}) => ReactElement;
+    Navigation?: (props: NavigationProps) => ReactElement;
+  };
+};
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:158]: This should be exporting the value, not the type.

**Review by timleslie — CHANGES_REQUESTED** [review]

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 37]
```diff
@@ -0,0 +1,344 @@
...
+    <NavigationContainer authenticatedItem={authenticatedItem}>
+      {/* ... */}
+    </NavigationContainer>
+  )
+}
+```
+
+### ListNavItems
+
+This component takes the provided Array of `lists` and renders a list of `NavItems`.
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:238]:
```suggestion
This component takes the provided array of `lists` and renders a list of `NavItems`.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 38]
```diff
@@ -0,0 +1,344 @@
...
+```
+
+As mentioned above the lists object has the following type:
+
+```typescript
+type lists = ListMeta[]
+type include = string[]
+```
+
+If an include prop is supplied, the component will only render out lists that match the list `keys` specified in the include array.
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:251]:
```suggestion
If an `include` prop is supplied, the component will only render out lists that match the list `keys` specified in the `include` array.
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 39]
```diff
@@ -0,0 +1,344 @@
...
+
+### ListNavItems
+
+This component takes the provided Array of `lists` and renders a list of `NavItems`.
+
+```tsx
+import { ListNavItems } from '@keystone-next/keystone/admin-ui/components'
+```
+
+As mentioned above the lists object has the following type:
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:244]:
```suggestion
The `lists` object has the following type:
```

**Review by timleslie — CHANGES_REQUESTED** [review]

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 40]
```diff
@@ -0,0 +1,301 @@
...
+import { NavigationContainer, NavItem, ListNavItems } from '@keystone-next/keystone/admin-ui/components';
+import type { NavigationProps } from '@keystone-next/keystone/admin-ui/components';
+
+export function CustomNavigation({ authenticatedItem, lists }: NavigationProps) {
+  return (
+    <NavigationContainer authenticatedItem={authenticatedItem}>
+      <NavItem href="/">Dashboard</NavItem>
+      <ListNavItems lists={lists}/>
+      <NavItem href="https://keystonejs.com/">
+        KeystoneJS Docs
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:117]:
```suggestion
        Keystone Docs
```

File: docs/pages/docs/guides/custom-admin-ui-navigation.mdx [hunk 41]
```diff
@@ -0,0 +1,301 @@
...
```
Comment (timleslie) [inline docs/pages/docs/guides/custom-admin-ui-navigation.mdx:143]:
```suggestion
        Keystone Docs
```

**Review by timleslie — CHANGES_REQUESTED** [review]

**Review by timleslie — APPROVED** [review]
Let's 🚢 it!

**Pull request merged**